core/services/ocr2/plugins/ocr2keeper/evm21/mercury/streams/streams.go

```diff
@@ -112,15 +110,15 @@ func (s *streams) Lookup(ctx context.Context, checkResults []ocr2keepers.CheckRe
 }
 
 // buildResult checks if the upkeep is allowed by Mercury and builds a streams lookup request from the check result
-func (s *streams) buildResult(ctx context.Context, i int, checkResult ocr2keepers.CheckResult, checkResults []ocr2keepers.CheckResult, lookups map[int]*mercury.StreamsLookup) {
+func (s *streams) buildResult(ctx context.Context, i int, checkResults []ocr2keepers.CheckResult, lookups map[int]*mercury.StreamsLookup) {
```

Comment:
in this function, we only ever use `checkResults[i]` - if this is the case, why not just pass a pointer `checkResult *ocr2keepers.CheckResult` instead of  the whole slice? That feels cleaner to me imho.